src/home/classes/Search.ts

```diff
@@ -0,0 +1,9 @@
+class Search {
+  public value = '';
+
+  constructor(newValue: string) {
+    this.value = newValue;
+  }
+}
+
+export default Search;
```

Comment:
Só por curiosidade, o componente será em classe mesmo mano?

Reply:
Eu to pensando em trabalhar com classe pra gerar components somente pra usar essa vantagem do TS melhor. Isso nada interfere no funcionament do React, somente na criação de objetos que vão ser enviados ou manipulados.